### ISSUE TITLE
Keras Custom Layer

### ISSUE THREAD

**Issue by VamshiG** [deeplearning4j/deeplearning4j#9861]
#### Issue Description

I have a keras custom dense layer developed using python.

        class CustomDense(tf.keras.layers.Layer):
            def __init__(self, num_units, activation = "sigmoid", **kwargs):
                super(CustomDense, self).__init__()
        
                self.num_units = num_units
                self.activation = Activation(activation)
                super(CustomDense, self).__init__(**kwargs)
                
            def get_config(self):
                config = super(CustomDense, self).get_config()
                config.update({"num_units": self.num_units, "activation" : self.activation})
                return config
        
            def build(self, input_shape):
        
                self.weight = self.add_weight(shape = [128, self.num_units], dtype = 'float32', name = 'other_features_weights')
                self.weight1 = self.add_weight(shape = [1, self.num_units], initializer = tf.keras.initializers.Constant(1.),
                                         dtype = 'float32',
                                         trainable = False, name = 'training_bid_weights')
           
                self.bias = self.add_weight(shape = [self.num_units], name = 'bias_weights')
        
            def call(self, input):
                self.final_weight = tf.concat((self.weight1, self.weight), axis = 0)                                
                y = tf.matmul(input, self.final_weight) + self.bias
                y = self.activation(y)
                return y

I am having issues in creating above custom dense layer in java.

If anyone knows how to do it , it would help me a lot

**Note: I am new to java

**Comment by agibsonccc** [comment]
@VamshiG  Custom layers can be imported like this: https://github.com/deeplearning4j/deeplearning4j/blob/e2b92619b299e13a181791bf5ecd53304cb393e0/platform-tests/src/test/java/org/eclipse/deeplearning4j/frameworkimport/keras/configurations/DeepCTRLambdaTest.java

Please post on our forums for further support that are not bugs. Thanks!
https://community.konduit.ai/

**Comment by VamshiG** [comment]
I have initialised weights like below in java :

        IWeightInit init = KerasInitilizationUtils.getWeightInitFromConfig(layerConfig, conf.getINIT_GLOROT_UNIFORM(),
                        true, conf, kerasMajorVersion);
        
        IWeightInit init1 = KerasInitilizationUtils.getWeightInitFromConfig(layerConfig, conf.getINIT_CONSTANT(),
                false, conf, kerasMajorVersion);
        
for python code  :

        self.weight = self.add_weight(shape = [128, self.num_units], dtype = 'float32', name = 'other_features_weights')
        self.weight1 = self.add_weight(shape = [1, self.num_units], initializer = tf.keras.initializers.Constant(1.),
                                     dtype = 'float32',
                                     trainable = False, name = 'training_bid_weights')
   
Now i need to concat this weights like :

     self.final_weight = tf.concat((self.weight1, self.weight), axis = 0) 

How do i do that in java?

Thanks for the help

**Comment by agibsonccc** [comment]
@VamshiG please do as I ask and move to the forums. More people with similar issues will find it in the future via search. I'll provide further help on how to set the layers up there. Thanks!

**Comment by VamshiG** [comment]
Sure will do 
Thanks
